### PR TITLE
Fix switching to heads on MSI MS-7D25 platforms

### PR DESCRIPTION
**Pull request by TomaszAIR**

### CODE REVIEW

File: meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh [hunk 1]
```diff
@@ -320,14 +320,24 @@ board_config() {
               HAVE_HEADS_FW="true"
               HEADS_REL_VER_DES="0.9.0"
               HEADS_LINK_DES="${FW_STORE_URL_DES}/MS-7D25/v${DASHARO_REL_VER_DES}/${DASHARO_REL_NAME}_v${DASHARO_REL_VER_DES}_ddr4_heads.rom"
-              HEADS_SWITCH_FLASHROM_OPT_OVERRIDE="" # we have to do full flashing
+              if [ ! -f $SE_credential_file ]; then
```
Comment (miczyg1) [inline meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh:323]: Shouldn't it be DES_IS_LOGGED?

File: meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh [hunk 2]
```diff
@@ -358,14 +368,24 @@ board_config() {
               HAVE_HEADS_FW="true"
               HEADS_REL_VER_DES="0.9.0"
               HEADS_LINK_DES="${FW_STORE_URL_DES}/MS-7D25/v${DASHARO_REL_VER_DES}/${DASHARO_REL_NAME}_v${DASHARO_REL_VER_DES}_ddr5_heads.rom"
-              HEADS_SWITCH_FLASHROM_OPT_OVERRIDE="" # we have to do full flashing
+              if [ ! -f $SE_credential_file ]; then
```
Comment (miczyg1) [inline meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh:371]: A/A

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Comment by miczyg1** [comment]
Also there is a bug here:

```
get_signing_keys() {
    local platform_keys=$PLATFORM_SIGN_KEY
    echo -n "Getting platform specific GPG key... "
    for key in $platform_keys; do
        wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$PLATFORM_SIGN_KEY -O - | gpg --import - >> $ERR_LOG_FILE 2>&1
        error_check "Cannot get $key key to verify signatures."
    done
    print_green "Done"
}
```

In the for loop we download `wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$PLATFORM_SIGN_KEY` instead of `wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$key`

File: meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh (hunk 1 above)
Comment (TomaszAIR) [reply meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh]: @miczyg1 you are right, fixed https://github.com/Dasharo/meta-dts/pull/104/commits/ec3c044f3da2b54dde55d89380cc00c5c48f34f6

**Comment by TomaszAIR** [comment]
> Also there is a bug here:
> 
> ```
> get_signing_keys() {
>     local platform_keys=$PLATFORM_SIGN_KEY
>     echo -n "Getting platform specific GPG key... "
>     for key in $platform_keys; do
>         wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$PLATFORM_SIGN_KEY -O - | gpg --import - >> $ERR_LOG_FILE 2>&1
>         error_check "Cannot get $key key to verify signatures."
>     done
>     print_green "Done"
> }
> ```
> 
> In the for loop we download `wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$PLATFORM_SIGN_KEY` instead of `wget -q https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/$key`

fixed in https://github.com/Dasharo/meta-dts/pull/104/commits/2621c17ab2edee54ea8f8fbd97bf11968ff73a99

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**